fusion_io/m3dc1_source.cpp

```diff
@@ -64,6 +64,7 @@ int m3dc1_source::open(const char* filename)
 int m3dc1_source::allocate_search_hint(void** s)
 {
   *s = new int;
```

Comment:
FYI, you can call the constructor directly in the new statement.
`*s = new int(-1)`